src/components/Form.js

```diff
@@ -26,7 +26,7 @@ export default class Form extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState(this.getStateFromProps(nextProps));
+    this.state = {...this.state, ...this.getStateFromProps(nextProps)};
```

Comment:
I just did some investigation an found that "setSate" in "componentWillReceiveProps" will actually not trigger render.
And the [react document](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops) said 
>  you may compare this.props and nextProps and perform state transitions using this.setState() in this method.
So I think the root cause was just the setState in onChange.
I will change this code back.